File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/GroupContactsUpdatedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.Group;
import com.shareyourproxy.api.domain.model.User;

import java.util.List;

/**
 * Created by Evan on 6/10/15.
 */
public class GroupContactsUpdatedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        GroupContactsUpdatedEventCallback.class.getClassLoader();
    public static final Creator<GroupContactsUpdatedEventCallback> CREATOR =
        new Creator<GroupContactsUpdatedEventCallback>() {
            @Override
            public GroupContactsUpdatedEventCallback createFromParcel(Parcel in) {
                return new GroupContactsUpdatedEventCallback(
                    (User) in.readValue(CL), (String) in.readValue(CL),
                    (List<Group>) in.readValue(CL));
            }

            @Override
            public GroupContactsUpdatedEventCallback[] newArray(int size) {
                return new GroupContactsUpdatedEventCallback[size];
            }
        };
    public final List<Group> contactGroups;
    public final String contactId;

    public GroupContactsUpdatedEventCallback(User user, String contactId, List<Group> groups) {
        super(user);
        this.contactId = contactId;
        this.contactGroups = groups;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(contactId);
        dest.writeValue(contactGroups);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/GroupContactsUpdatedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.Group
import com.shareyourproxy.api.domain.model.User

/**
 * Created by Evan on 6/10/15.
 */
class GroupContactsUpdatedEventCallback(user: User, val contactId: String, val contactGroups: List<Group>) : UserEventCallback(user) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(contactId)
        dest.writeValue(contactGroups)
    }

    companion object {
        private val CL = GroupContactsUpdatedEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<GroupContactsUpdatedEventCallback> = object : Parcelable.Creator<GroupContactsUpdatedEventCallback> {
            override fun createFromParcel(parcel: Parcel): GroupContactsUpdatedEventCallback {
                return GroupContactsUpdatedEventCallback(parcel.readValue(CL) as User, parcel.readValue(CL) as String, parcel.readValue(CL) as List<Group>)
            }

            override fun newArray(size: Int): Array<GroupContactsUpdatedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
